endpointslice/endpointslice_controller.go

```diff
@@ -106,6 +106,7 @@ func NewControllerWithName(ctx context.Context, podInformer coreinformers.PodInf
 	endpointUpdatesBatchPeriod time.Duration,
 	controllerName string,
 	checkService func(service *v1.Service) bool,
+	noServiceHook func(namespace, name string) error,
```

Comment:
Just one small suggestion, can we change the function name to be `notFoundServiceHook`?

Reply:
Sure 👍, I just did this!